### PR TITLE
GH#220 - Softmax Utils Function

### LINKED ISSUE #220: Move Softmax calculation to tested utility function
Also align with implementations like tensorflow.

Working on a PR for this.

### PR DESCRIPTION
**Pull request by luckylwk**
Closes #220 

For review: @BruceYanghy, @XiaoYangLiu-FinRL

### CODE REVIEW

**Comment by zhumingpassional** [comment]
this branch has conflicts with master. 

could  you pls describe the reason why using the softmax you have written?
